fineract-loan/src/main/java/org/apache/fineract/accounting/producttoaccountmapping/serialization/ProductToGLAccountMappingFromApiJsonDeserializer.java

```diff
@@ -84,7 +84,7 @@ public void validateForLoanProductCreate(final String json) {
         final Integer accountingRuleType = this.fromApiJsonHelper.extractIntegerNamed("accountingRule", element, Locale.getDefault());
         baseDataValidator.reset().parameter("accountingRule").value(accountingRuleType).notNull().inMinMaxRange(1, 4);
 
-        if (isCashBasedAccounting(accountingRuleType) || isAccrualBasedAccounting(accountingRuleType)) {
+        if (AccountingValidations.isCashBasedAccounting(accountingRuleType)) {
```

Comment:
Are you sure about this change?

Reply:
I dont mind to put this logic into a new class (however a not static one would be better, because it can be overridden if needed), but the original condition said: IF it is cash based on accrual based, but your validator is checkint only if it is cash based. Are you sure this change is correct?